SharedSources/MediaLibraryService.swift

```diff
@@ -284,10 +284,25 @@ extension MediaLibraryService {
 
     func fetchMedia(with mrl: URL?) -> VLCMLMedia? {
         guard let mrl = mrl  else {
+            assertionFailure("no mrl")
             return nil
         }
         return medialib.media(withMrl: mrl)
     }
+
+    func savePlaybackState(from player: VLCPlaybackController) {
+
+        let media = player.currentlyPlayingMedia
+        let mlMedia = fetchMedia(with: media.url.absoluteURL)!
+
+        mlMedia.isNew = false
+        mlMedia.progress = player.playbackPosition
+        mlMedia.audioTrackIndex = player.indexOfCurrentAudioTrack
+        mlMedia.subtitleTrackIndex = player.indexOfCurrentSubtitleTrack
+        mlMedia.chapterIndex = player.indexOfCurrentChapter
+        mlMedia.titleIndex = player.indexOfCurrentTitle
+        //create a new thumbnail
```

Comment:
the old code triggered a thumbnail update with the current position not sure if this is doable with the new medialibrary

Reply:
Indeed, related ticket [here](https://code.videolan.org/videolan/medialibrary/issues/82).